### PR TITLE
level brightness feature and ability to disable menu background

### PR DESCRIPTION
**Pull request by rfomin**
Feature request: [[doomworld]](https://www.doomworld.com/forum/topic/112333-this-is-woof-1030-sep-23-2022-updated-winmbf/?do=findComment&comment=2560143)

Taken from International Doom and Unity port. I believe this option is more convenient without menu background.

### CODE REVIEW

**Comment by liPillON** [comment]
would it be possible to have a third choice for the "Draw Menu Background" option?
- Background On
- Background Off
- Replace Background with dark tint

**Comment by rfomin** [comment]
> would it be possible to have a third choice for the "Draw Menu Background" option?

Done in https://github.com/fabiangreffrath/woof/pull/778/commits/6fe40e8014eff3d570dfdf792674ef20c7d70a06

File: src/m_menu.c [hunk 1]
```diff
@@ -4034,14 +4049,15 @@ setup_menu_t gen_settings2[] = { // General Settings screen2
   {"Solid Status Bar Background", S_YESNO, m_null, M_X,
    M_Y + general_solidbackground*M_SPC, {"st_solidbackground"}},
 
-  {"", S_SKIP, m_null, M_X, M_Y + general_stub2*M_SPC},
+  {"Level Brightness", S_THERMO, m_null, M_X_THRM,
```
Comment (fabiangreffrath) [inline src/m_menu.c:4052]: This is a bit more than a "Display Option", right? Isn't this plain cheating in certain (very dark) maps?

Comment (rfomin) [reply src/m_menu.c:4052]: Yes, I disabled it in `strictmode`. We can move it to another menu group, maybe QOL?

Comment (fabiangreffrath) [reply src/m_menu.c:4052]: Yes, please. I see it rather in the "disable power-up flashes" category.

File: src/m_menu.c [hunk 2]
```diff
@@ -3996,6 +4007,10 @@ static const char *death_use_action_strings[] = {
   "default", "last save", "nothing", NULL
 };
 
+static const char *menu_background_strings[] = {
+  "on", "off", "dark tint", NULL
```
Comment (fabiangreffrath) [inline src/m_menu.c:4011]: Does "dim" fit better than "dark tint"? What do the native speakers say?

File: src/m_menu.c [hunk 3]
```diff
@@ -6441,6 +6473,16 @@ void M_StartControlPanel (void)
 
 void M_Drawer (void)
 {
+   if (setup_active && menu_background == 2)
+   {
+      int y;
+      byte *dest = screens[0];
```
Comment (fabiangreffrath) [inline src/m_menu.c:6479]: When I still had this in Crispy, it had a nice fade out effect. Just an idea. 

https://github.com/fabiangreffrath/crispy-doom/commit/cd807c0b5f76f5b09f119ff1bd8e463824a61bca

File: src/m_menu.c (hunk 2 above)
Comment (mikeday0) [reply src/m_menu.c:4017]: "Dim" is better than "dark tint" for me, but I do not have a sense of how widespread that word is for non-native speakers. A simple "dark" might also get the job done here.

File: src/m_menu.c (hunk 3 above)
Comment (rfomin) [reply src/m_menu.c:6481]: I also like the shadowed text effect, why did you remove it? I think we can add it to Woof and bring it back to Crispy.

File: src/f_finale.c [hunk 4]
```diff
@@ -38,6 +38,7 @@
 #include "d_deh.h"  // Ty 03/22/98 - externalizations
 #include "m_misc2.h" // [FG] M_StringDuplicate()
 #include "m_swap.h"
+#include "r_draw.h"
```
Comment (fabiangreffrath) [inline src/f_finale.c:41]: Do we still need to `#include m_menu.h` here?

File: src/m_menu.c [hunk 5]
```diff
@@ -4072,6 +4036,9 @@ setup_menu_t gen_settings3[] = { // General Settings screen3
   {"Screen melt", S_YESNO, m_null, M_X,
    M_Y + general_screen_melt*M_SPC, {"screen_melt"}},
 
+  {"Level Brightness", S_THERMO, m_null, M_X_THRM,
```
Comment (fabiangreffrath) [inline src/m_menu.c:4039]: How about "Extra Level Brightness"?

File: src/m_menu.c [hunk 6]
```diff
@@ -6441,6 +6416,28 @@ void M_StartControlPanel (void)
...
+      static int firsttic;
+
+      for (y = 0; y < (SCREENWIDTH << hires) * (SCREENHEIGHT << hires); y++)
+      {
+         dest[y] = colormaps[0][menushade * 256 + dest[y]];
+      }
+
+      if (menushade < 16 && gametic != firsttic)
+      {
+         menushade += 4;
```
Comment (fabiangreffrath) [inline src/m_menu.c:6434]: So, the whole fading takes only 4 tics, isn't this a bit short?

File: src/m_menu.c (hunk 5 above)
Comment (rfomin) [reply src/m_menu.c:4039]: That's what it was called in the Unity port and inter-doom, I think users have become familiar with this name.

File: src/m_menu.c (hunk 6 above)
Comment (rfomin) [reply src/m_menu.c:6434]: I like it when it fading faster. What do you suggest?

File: src/f_finale.c (hunk 4 above)
Comment (rfomin) [reply src/f_finale.c:40]: Fixed in https://github.com/fabiangreffrath/woof/pull/778/commits/6ed1d5d7a2401c698c64da2015c0d837d7f8b4fa

File: src/m_menu.c (hunk 6 above)
Comment (fabiangreffrath) [reply src/m_menu.c:6434]: > I like it when it fading faster. What do you suggest?

I like it when it takes a bit longer. 😛 😉

Comment (rfomin) [reply src/m_menu.c]: I guess 8 tics is compromise 😄

Comment (fabiangreffrath) [reply src/m_menu.c]: 💪

File: src/m_menu.c (hunk 3 above)
Comment (fabiangreffrath) [reply src/m_menu.c:6424]: > I also like the shadowed text effect, why did you remove it? I think we can add it to Woof and bring it back to Crispy.

It was deemed a pointless deviation from the Vanila experience. At this point, I decided that all UI that was already present in Vanilla should be preserved as far as possible, and only new ("added by Crispy") UI got some blink.

**Review by fabiangreffrath — APPROVED** [review]

**Pull request merged**